app/screens/settings/notification_mention/notification_mention.tsx

```diff
@@ -0,0 +1,286 @@
+                                </Text>
+                            )}
+                            description={(
+                                <FormattedText
+                                    id='notification_settings.mentions.sensitiveUsername'
+                                    defaultMessage='Your non-case sensitive username'
+                                />
+                            )}
+                            selected={usernameMention}
+                            action={toggleUsernameMention}
+                            actionType='toggle'
+                        />
+                    )}
+                    <View style={styles.separator}/>
+                    <SectionItem
+                        label={(
+                            <Text>
+                                {'@channel, @all, @here'}
+                            </Text>
+                        )}
+                        description={(
+                            <FormattedText
+                                id='notification_settings.mentions.channelWide'
+                                defaultMessage='Channel-wide mentions'
+                            />
+                        )}
+                        action={toggleChannelMentions}
+                        actionType='toggle'
+                        selected={channel}
+                    />
+                    <View style={styles.separator}/>
+                    <SectionItem
+                        label={(
+                            <FormattedText
+                                id='notification_settings.mentions.keywords'
+                                defaultMessage='Keywords'
+                            />
+                        )}
+                        description={mentionKeysComponent}
+                        action={goToNotificationSettingsMentionKeywords}
+                        actionType='arrow'
+                    />
+                </>
+            </Section>
+        );
+    };
+
+    const goToNotificationSettingsMentionKeywords = () => {
+        return Alert.alert(
+            'The functionality you are trying to use has not yet been implemented.',
+        );
+    };
+    const toggleChannelMentions = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                channel: !channel,
+            },
+        });
+    };
+    const toggleUsernameMention = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                usernameMention: !usernameMention,
+            },
+        });
+    };
+    const toggleFirstNameMention = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                firstName: !firstName,
+            },
+        });
+    };
+
+    const setReplyNotifications = (notifType: string) => {
+        setReplyNotificationType(notifType);
+    };
+
+    const renderReplySection = () => {
```

Comment:
same, this can be moved to its own component file

Reply:
done